elm-src/index.elm

```diff
@@ -0,0 +1,489 @@
+    button
+        (if cond then
+            [ onClick (ChangePage page) ]
+         else
+            [ disabled True ]
+        )
+        [ text txt ]
+
+
+view : Model -> Html Msg
+view model =
+    if List.length model.options > 0 then
+        viewOptions model
+    else
+        div [] [ h1 [] [ text (toString model.status) ] ]
+
+
+debounce =
+    Debounce.debounceCmd debounce_cfg
+
+
+debounce_cfg : Debounce.Config Model Msg
+debounce_cfg =
+    Debounce.config
+        .debounce_state
+        -- getState   : Model -> Debounce.State
+        (\model s -> { model | debounce_state = s })
+        -- setState   : Debounce.State -> Model -> Debounce.State
+        Bounce
+        -- msgWrapper : Msg a -> Msg
+        200
+
+
+
+-- timeout ms : Float
+
+
+viewOptions : Model -> Html Msg
+viewOptions model =
+    div []
+        [ p []
+            [ input [ id "search", class "search-query span3", autofocus True, value model.query, onInput ChangeQuery ] []
+            ]
+        , hr [] []
+        , p [ id "how-many" ]
+            [ em []
+                [ text
+                    (if (List.length model.matchingOptions) == 0 then
+                        "Showing no results"
+                     else
+                        (String.concat
+                            [ "Showing results "
+                            , (toString (((model.page - 1) * 15) + 1))
+                            , "-"
+                            , (toString (min (List.length model.matchingOptions) ((model.page) * 15)))
+                            , " of "
+                            , (toString (List.length model.matchingOptions))
+                            , "."
+                            ]
+                        )
+                    )
+                ]
+            ]
+        , table [ class "options table table-hover" ]
+            [ thead []
+                [ tr []
+                    [ th [] [ text "Option name" ]
+                    ]
+                ]
+            , tbody []
+                (List.concat (List.map (optionToTd model.selected) (fetch_page model.page model.matchingOptions)))
+            ]
+        , ul [ class "pager" ]
+            [ li []
+                [ (changePageIf (not (model.page <= 1)) 1 "« First")
+                ]
+            , li []
+                [ (changePageIf (model.page > 1) (model.page - 1) "‹ Previous")
+                ]
+            , li []
+                [ (changePageIf (model.page < (total_pages model.matchingOptions)) (model.page + 1) "Next ›")
+                ]
+            , li []
+                [ (changePageIf (model.page < (total_pages model.matchingOptions)) (total_pages model.matchingOptions) "Last »")
+                ]
+            ]
+        ]
+
+
+
+-- SUBSCRIPTIONS
+
+
+subscriptions : Model -> Sub Msg
+subscriptions model =
+    Sub.none
+
+
+
+-- HTTP
+
+
+getOptions : Cmd Msg
+getOptions =
+    Http.send FetchedOptions (Http.get "./options.json" decodeOptions)
```

Comment:
In this case (and quite a few others), I personally like using `<|` to remove some parents:

```
Http.send FetchedOptions (Http.get "./options.json" decodeOptions)
Http.send FetchedOptions <| Http.get "./options.json" decodeOptions
```

Reply:
Or, perhaps, even more sensical

```
Http.get "./options.json" decodeOptions
    |> Http.send FetchedOptions
``` 

i.e. build the request, then send it